helm-app-operator/pkg/helm/controller/reconcile.go

```diff
@@ -54,52 +53,60 @@ func (r HelmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.
 	o.SetGroupVersionKind(r.GVK)
 	o.SetNamespace(request.Namespace)
 	o.SetName(request.Name)
-	logrus.Debugf("Processing %s", util.ResourceString(o))
+	log := log.WithValues(
+		"namespace", o.GetNamespace(),
+		"name", o.GetName(),
+		"apiVersion", o.GetAPIVersion(),
+		"kind", o.GetKind(),
+	)
+	log.V(1).Info("Reconciling")
 
 	err := r.Client.Get(context.TODO(), request.NamespacedName, o)
 	if apierrors.IsNotFound(err) {
 		return reconcile.Result{}, nil
 	}
 	if err != nil {
-		logrus.Errorf("failed to lookup %s: %s", util.ResourceString(o), err)
+		log.Error(err, "failed to lookup resource")
 		return reconcile.Result{}, err
 	}
 
+	manager := r.ManagerFactory.NewManager(o)
+	status := types.StatusFor(o)
+	log = log.WithValues("release", manager.ReleaseName())
+
 	deleted := o.GetDeletionTimestamp() != nil
 	pendingFinalizers := o.GetFinalizers()
 	if !deleted && !contains(pendingFinalizers, finalizer) {
-		logrus.Debugf("Adding finalizer \"%s\" to %s", finalizer, util.ResourceString(o))
+		log.V(1).Info("Adding finalizer", "finalizer", finalizer)
 		finalizers := append(pendingFinalizers, finalizer)
 		o.SetFinalizers(finalizers)
 		err := r.Client.Update(context.TODO(), o)
 		return reconcile.Result{}, err
 	}
 
-	manager := r.ManagerFactory.NewManager(o)
-	status := types.StatusFor(o)
-	releaseName := manager.ReleaseName()
-
 	if err := manager.Sync(context.TODO()); err != nil {
-		logrus.Errorf("failed to sync release for %s release=%s: %s", util.ResourceString(o), releaseName, err)
+		log.Error(err, "failed to sync release")
 		return reconcile.Result{}, err
 	}
 
 	if deleted {
 		if !contains(pendingFinalizers, finalizer) {
-			logrus.Infof("Resource %s is terminated, skipping reconciliation", util.ResourceString(o))
+			log.Info("Resource is terminated, skipping reconciliation")
 			return reconcile.Result{}, nil
 		}
 
 		uninstalledRelease, err := manager.UninstallRelease(context.TODO())
 		if err != nil && err != release.ErrNotFound {
-			logrus.Errorf("failed to uninstall release for %s release=%s: %s", util.ResourceString(o), releaseName, err)
+			log.Error(err, "failed to uninstall release")
 			return reconcile.Result{}, err
 		}
 		if err == release.ErrNotFound {
-			logrus.Infof("Release %s for resource %s not found, removing finalizer", releaseName, util.ResourceString(o))
+			log.Info("Release not found, removing finalizer")
 		} else {
-			diff := util.Diff(uninstalledRelease.GetManifest(), "")
-			logrus.Infof("Uninstalled release for %s release=%s; diff:\n%s", util.ResourceString(o), releaseName, diff)
+			log.Info("Uninstalled release")
+			if log.Enabled() {
+				fmt.Println(util.Diff(uninstalledRelease.GetManifest(), ""))
```

Comment:
Printing if log is enabled?
Why not just always debug(V(1)) log this:
```Go
log.V(1).Info(...)
```

Reply:
So there are two issues to discuss with the diff logs actually. The first is that with the JSON formatter we're using, the diffs come out all on one line and impossible to read, hence the change to `fmt`.

The second, which is your question, is which level should the diffs be logged at? INFO or DEBUG? Up to this point, we've been logging them at the INFO level.

Reply:
Yeah I think Uninstall can be logged as Info since it's not a common operation.

There might be a way to split up the Diff output into multiple lines although they might be across multiple log lines. Which wouldn't be very nice to look at. We can keep this as fmt for now.

Wondering how it currently looks with the other logs though? Have you tried out a sample output.